chain/indexer/src/streamer/streamer.rs

```diff
@@ -84,39 +134,94 @@ async fn fetch_block_by_height(
         .map_err(|err| FailedToFetchData::String(err))
 }
 
-/// This function supposed to return the entire `BlockResponse`.
-/// It fetches the block and fetches all the chunks for the block
+/// This function supposed to return the entire `StreamerMessage`.
+/// It fetches the block and all related parts (chunks, outcomes, state changes etc.)
 /// and returns everything together in one struct
-async fn fetch_block_response(
+async fn build_streamer_message(
     client: &Addr<near_client::ViewClientActor>,
     block: views::BlockView,
-    outcomes_to_get: impl Iterator<Item = types::TransactionOrReceiptId>,
-) -> Result<FetchBlockResponse, FailedToFetchData> {
+    db: &DB,
+) -> Result<StreamerMessage, FailedToFetchData> {
     let chunks = fetch_chunks(&client, &block.chunks).await?;
-    let (outcomes, mut outcomes_to_retry) = fetch_outcomes(&client, outcomes_to_get).await;
+
+    let mut outcomes_to_fetch: Vec<TransactionOrReceiptId> = vec![];
+    let mut local_receipts: Vec<views::ReceiptView> = vec![];
+    let mut outcomes: Vec<Outcome> = vec![];
 
     for chunk in &chunks {
-        outcomes_to_retry.extend(chunk.transactions.iter().map(|transaction| {
-            types::TransactionOrReceiptId::Transaction {
-                transaction_hash: transaction.hash.clone(),
-                sender_id: transaction.signer_id.to_string(),
+        for transaction in &chunk.transactions {
+            match client
+                .send(near_client::GetExecutionOutcome {
+                    id: types::TransactionOrReceiptId::Transaction {
+                        transaction_hash: transaction.hash.clone(),
+                        sender_id: transaction.signer_id.to_string(),
+                    },
+                })
+                .await
+                .map_err(|e| eprintln!("{:?}", e))
```

Comment:
We should not print errors. Format it as a string here, if you want, and log/warn/err in the matching branch.
```suggestion
                .map_err(|e| format!("Failed to fetch execution outcome: {:?}", e))
```